scheduler/pkg/kafka/gateway/consumer.go

```diff
@@ -269,6 +290,7 @@ func (kc *InferKafkaConsumer) Serve() {
 		select {
 		case <-kc.done:
 			logger.Infof("Stopping")
+			kc.running.Store(false)
```

Comment:
Related to my comment about a producer in a consumer struct, there are two running flags that confused me initially thinking they should be only one of them. At least rename to reflect what's the flag concern?

Reply:
will rename to producerActive